classic_cp/N_Queens/N_Queens.jl

```diff
@@ -0,0 +1,182 @@
+using SeaPearl
+
+struct OutputDataQueens
+    nb_sols ::Int
+    indices ::Matrix{Int}
+end
+
+"""
+    model_queens(board_size::Int; benchmark=false, variableSelection=SeaPearl.MinDomainVariableSelection{false}(), valueSelection=SeaPearl.BasicHeuristic())
+
+return the SeaPearl model for to the N-Queens problem, using SeaPearl.MinDomainVariableSelection heuristique
+and  SeaPearl.AllDifferent (without solving it)
+
+# Arguments
+- `board_size::Int`: dimension of the board
+- 'variableSelection': SeaPearl variable selection. By default: SeaPearl.MinDomainVariableSelection{false}()
+- 'valueSelection': SeaPearl value selection. By default: =SeaPearl.BasicHeuristic()
+"""
+function model_queens(board_size::Int; benchmark=false, variableSelection=SeaPearl.MinDomainVariableSelection{false}(), valueSelection=SeaPearl.BasicHeuristic())
+    trailer = SeaPearl.Trailer()
+    model = SeaPearl.CPModel(trailer)
+
+    rows = Vector{SeaPearl.AbstractIntVar}(undef, board_size)
+    for i = 1:board_size
+        rows[i] = SeaPearl.IntVar(1, board_size, "row_"*string(i), trailer)
+        SeaPearl.addVariable!(model, rows[i]; branchable=true)
+    end
+
+    rows_plus = Vector{SeaPearl.AbstractIntVar}(undef, board_size)
+    for i = 1:board_size
+        rows_plus[i] = SeaPearl.IntVarViewOffset(rows[i], i, rows[i].id*"+"*string(i))
```

Comment:
I think row_plus and row_minus are the diagonals. 
I would rename the variables, but it is just a suggestion.